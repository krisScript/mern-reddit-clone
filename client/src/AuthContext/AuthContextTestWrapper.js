import {AuthContextData} from './AuthContext'
import {BrowserRouter} from 'react-router-dom'
import React from 'react'
const AuthContextTestWrapper = ({authState,children,updateUserDataReducer}) => {
    return (
       <AuthContextData.Provider value={{authState,updateUserDataReducer}}>
        <BrowserRouter>
          {children}
          </BrowserRouter>
        </AuthContextData.Provider>
    )
}
export default AuthContextTestWrapper 