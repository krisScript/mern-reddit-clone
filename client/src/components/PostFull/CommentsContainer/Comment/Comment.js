import React from 'react'
const Comment= props => {
    const {comment} = props
    const {content,author,upvotes,downvotes,creationDate} = comment
    console.log(creationDate)
    console.log(comment)
    const dateObject = new Date(Date.parse(creationDate));
  const dateReadable = dateObject.toDateString();
  console.log(dateReadable)
    return (
        <div className="comment">
        <div className="comment-details">
        <p className="comment-author">{author}</p>
            <p className="comment-date">{dateReadable}</p>
        </div>
            <h5 className="comment-content">{content}</h5>
        </div>
    )
}
export default Comment