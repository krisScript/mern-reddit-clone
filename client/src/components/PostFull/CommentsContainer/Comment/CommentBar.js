import React,{useState,useContext} from 'react'
import deleteComment from './deleteComment'
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { AuthContextData } from '../../../../AuthContext/AuthContext';
import {
  faShare,faCommentAlt, faBookmark,faTrashAlt,faPen,
} from '@fortawesome/free-solid-svg-icons';
library.add(faShare,faCommentAlt,faBookmark,faTrashAlt,faPen,);
const CommentBar= props => {
    const {authState} = useContext(AuthContextData)
    const {userId,token} = authState
    const {authorId,comments,deleteCommentElement,commentId,toggle,setEditComment,comment,on} = props
    const deleteHandler = async () => {
        console.log(authState)
        const data = await deleteComment(commentId,token)
        if(data.msg === 'Comment Deleted'){
            deleteCommentElement(commentId)
        }
    }
    const editHandler = () => {
        if(on){

        }else {
            toggle()
            setEditComment(comment)
        }
    }
    let autorizedContent = ''
     if(authorId === userId){
        autorizedContent = 
        <>
         <button onClick={deleteHandler} className="button post-info-button">
        <FontAwesomeIcon icon="trash-alt" /><span>Delete</span>
        </button>
        <button onClick={editHandler} className="button post-info-button">
        <FontAwesomeIcon icon="pen" /><span>Edit</span>
        </button>
        </>
    }
    
    return (
        <div>
            <div className="post-info">
        <button className="button post-info-button">
        <FontAwesomeIcon icon="share" /><span>Share</span>
        </button>
        <button className="button post-info-button">
        <FontAwesomeIcon icon="comment-alt" /><span>{comments}</span>
        </button>
        <button className="button post-info-button">
        <FontAwesomeIcon icon="bookmark" /><span>Save</span>
        </button>
        {autorizedContent}
        </div>
        </div>
    )
}
export default CommentBar