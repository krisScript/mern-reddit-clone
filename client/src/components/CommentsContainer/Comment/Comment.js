import React from 'react';
import CommentBar from '../CommentBar/CommentBar';
import PropTypes from 'prop-types';
import commentType from '../../PropTypes/commentType'
const Comment = ({
  comment,
  deleteCommentElement,
  toggle,
  setEditComment,
  on
}) => {
  const {
    content,
    author,
    upvotes,
    downvotes,
    creationDate,

  } = comment;
  const dateObject = new Date(Date.parse(creationDate));
  const dateReadable = dateObject.toDateString();
  return (
    <div className="comment">
      <div className="comment-details">
        <p className="comment-author">u/{author}</p>
        <p className="comment-date">{dateReadable}</p>
      </div>
      <h5 className="comment-content">{content}</h5>
      <CommentBar
        deleteCommentElement={deleteCommentElement}
        toggle={toggle}
        setEditComment={setEditComment}
        comment={comment}
        on={on}
      />
    </div>
  );
};
Comment.propTypes = {
  comment:commentType.isRequired,
  deleteCommentElement: PropTypes.func.isRequired,
  toggle: PropTypes.func.isRequired,
  setEditComment: PropTypes.func.isRequired,
  on: PropTypes.bool.isRequired
};

export default Comment;
