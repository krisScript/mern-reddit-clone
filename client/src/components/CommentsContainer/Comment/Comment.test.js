import React from 'react';
import { render, fireEvent, waitForElement } from 'react-testing-library';
import Comment from './Comment';
import AuthContextTestWrapper from '../../../AuthContext/AuthContextTestWrapper'
const updateUserDataReducer = jest.fn()
const authState =  {isAuth:false,token:null,userId:null,userData:null,updateUserDataReducer}

describe('<Comment/>',() => {
    const comment = {
        creationDate :'2019-04-01T13:51:12.885Z',
        communityId :'testId',
        authorId :"authorId",
        author :"Joseph Joestar",
        postId :'testPostId',
        postTitle: 'tstPostTitle',
        content :'Your next line is...',
        upvotes :10,
        downvotes :3,
        comments:1
    }
    const toggle = jest.fn()
    const deleteCommentElement  = jest.fn()
    const setEditComment  = jest.fn()
    const on = false
    const humanReadableCreationDate = 'Mon Apr 01 2019'
    
    const { container, rerender,queryByPlaceholderText,queryByValue ,getByPlaceholderText,getByText } = render (<Comment comment={comment} toggle={toggle} deleteCommentElement={deleteCommentElement} setEditComment={setEditComment} on={on} />,{wrapper: ({ children }) => (
      <AuthContextTestWrapper authState={authState} children={children} />
      )})
      it('snapshot',() => {
        expect(container).toMatchSnapshot()
    })

      it('should have author element',  () => {
        const element =  getByText(`u/${comment.author}`)
        expect(element.className).toMatch('comment-author')
        expect(element).toBeTruthy;
      });
      it('should have date element',  () => {
        const element =  getByText(humanReadableCreationDate)
        expect(element.className).toMatch('comment-date')
        expect(element).toBeTruthy;
      });
      it('should have content element',  () => {
        const element =  getByText(`${comment.content}`)
        expect(element.className).toMatch('comment-content')
        expect(element).toBeTruthy;
      });
      
})

