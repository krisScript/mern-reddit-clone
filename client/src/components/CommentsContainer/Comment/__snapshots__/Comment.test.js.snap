// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment/> snapshot 1`] = `
<div>
  <div
    class="comment"
  >
    <div
      class="comment-details"
    >
      <p
        class="comment-author"
      >
        u/
        Joseph Joestar
      </p>
      <p
        class="comment-date"
      >
        Mon Apr 01 2019
      </p>
    </div>
    <h5
      class="comment-content"
    >
      Your next line is...
    </h5>
    <div>
      <div
        class="post-info"
      >
        <button
          class="button post-info-button"
          data-testid="share-button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-share fa-w-16 "
            data-icon="share"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M503.691 189.836L327.687 37.851C312.281 24.546 288 35.347 288 56.015v80.053C127.371 137.907 0 170.1 0 322.326c0 61.441 39.581 122.309 83.333 154.132 13.653 9.931 33.111-2.533 28.077-18.631C66.066 312.814 132.917 274.316 288 272.085V360c0 20.7 24.3 31.453 39.687 18.164l176.004-152c11.071-9.562 11.086-26.753 0-36.328z"
              fill="currentColor"
            />
          </svg>
          <span>
            Share
          </span>
        </button>
        <button
          class="button post-info-button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-comment-alt fa-w-16 "
            data-icon="comment-alt"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M448 0H64C28.7 0 0 28.7 0 64v288c0 35.3 28.7 64 64 64h96v84c0 9.8 11.2 15.5 19.1 9.7L304 416h144c35.3 0 64-28.7 64-64V64c0-35.3-28.7-64-64-64z"
              fill="currentColor"
            />
          </svg>
          <span>
            1
          </span>
        </button>
        <button
          class="button post-info-button"
          data-testid="save-button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-bookmark fa-w-12 "
            data-icon="bookmark"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z"
              fill="currentColor"
            />
          </svg>
          <span>
            Save
          </span>
        </button>
        
      </div>
    </div>
  </div>
</div>
`;
